File: java/src/main/java/com/allstask/api/BatchJobs.java
// this file is @generated
package com.allstask.api;

import com.allstask.AllstaskHttpClient;
import com.allstask.Utils;
import com.allstask.exceptions.ApiException;
import com.allstask.models.BatchJobDetailResponse;
import com.allstask.models.BatchJobFailuresResponse;
import com.allstask.models.BatchJobListResponse;

import okhttp3.HttpUrl;

import java.io.IOException;

public class BatchJobs {
    private final AllstaskHttpClient client;

    public BatchJobs(AllstaskHttpClient client) {
        this.client = client;
    }

    /** List batch jobs with optional filtering by type and status. */
    public BatchJobListResponse listBatchJobs() throws IOException, ApiException {

        return this.listBatchJobs(new BatchJobsListBatchJobsOptions());
    }

    /** List batch jobs with optional filtering by type and status. */
    public BatchJobListResponse listBatchJobs(final BatchJobsListBatchJobsOptions options)
            throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/batch-jobs");
        if (options.jobType != null) {
            url.addQueryParameter("job_type", Utils.serializeQueryParam(options.jobType));
        }
        if (options.status != null) {
            Utils.addExplodedQueryParameter(url, "status", options.status);
        }
        if (options.page != null) {
            url.addQueryParameter("page", Utils.serializeQueryParam(options.page));
        }
        if (options.perPage != null) {
            url.addQueryParameter("per_page", Utils.serializeQueryParam(options.perPage));
        }
        return this.client.executeRequest(
                "GET", url.build(), null, null, BatchJobListResponse.class);
    }

    /** Retrieve a single batch job with its chunks and failures. */
    public BatchJobDetailResponse getBatchJob(final String batchJobId)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/batch-jobs/%s", batchJobId));
        return this.client.executeRequest(
                "GET", url.build(), null, null, BatchJobDetailResponse.class);
    }

    /** Retrieve paginated failures for a batch job. */
    public BatchJobFailuresResponse listBatchJobFailures(final String batchJobId)
            throws IOException, ApiException {
        return this.listBatchJobFailures(batchJobId, new BatchJobsListBatchJobFailuresOptions());
    }

    /** Retrieve paginated failures for a batch job. */
    public BatchJobFailuresResponse listBatchJobFailures(
            final String batchJobId, final BatchJobsListBatchJobFailuresOptions options)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/batch-jobs/%s/failures", batchJobId));
        if (options.chunkId != null) {
            url.addQueryParameter("chunk_id", Utils.serializeQueryParam(options.chunkId));
        }
        if (options.limit != null) {
            url.addQueryParameter("limit", Utils.serializeQueryParam(options.limit));
        }
        if (options.offset != null) {
            url.addQueryParameter("offset", Utils.serializeQueryParam(options.offset));
        }
        return this.client.executeRequest(
                "GET", url.build(), null, null, BatchJobFailuresResponse.class);
    }
}

File: typescript/src/api/batchJobs.ts
// this file is @generated
import {
    BatchJobChunkId,
    BatchJobChunkIdSerializer,
} from '../models/batchJobChunkId';
import {
    BatchJobDetailResponse,
    BatchJobDetailResponseSerializer,
} from '../models/batchJobDetailResponse';
import {
    BatchJobFailuresResponse,
    BatchJobFailuresResponseSerializer,
} from '../models/batchJobFailuresResponse';
import {
    BatchJobListResponse,
    BatchJobListResponseSerializer,
} from '../models/batchJobListResponse';
import {
    BatchJobStatus,
    BatchJobStatusSerializer,
} from '../models/batchJobStatus';
import {
    BatchJobType,
    BatchJobTypeSerializer,
} from '../models/batchJobType';
import { HttpMethod, AllstaskRequest, AllstaskRequestContext } from "../request";

export interface BatchJobsListBatchJobsOptions {
        jobType?: BatchJobType;
        status?: BatchJobStatus[];
        /** Page number (0-indexed) */
            page?: number;
        /** Number of items per page */
            perPage?: number;
        }

    export interface BatchJobsListBatchJobFailuresOptions {
        chunkId?: BatchJobChunkId;
        limit?: number;
        offset?: number;
        }

    export class BatchJobs {
    public constructor(private readonly requestCtx: AllstaskRequestContext) {}

    /** List batch jobs with optional filtering by type and status. */
        public listBatchJobs(
            options?: BatchJobsListBatchJobsOptions,
            ): Promise<BatchJobListResponse> {
            const request = new AllstaskRequest(HttpMethod.GET, "/api/v1/batch-jobs");

            request.setQueryParam("job_type", options?.jobType);
            request.setQueryParam("status", options?.status);
            request.setQueryParam("page", options?.page);
            request.setQueryParam("per_page", options?.perPage);
            
                return request.send(
                    this.requestCtx,
                    BatchJobListResponseSerializer._fromJsonObject,
                );
            }

        

    /** Retrieve a single batch job with its chunks and failures. */
        public getBatchJob(
            batchJobId: string,
            ): Promise<BatchJobDetailResponse> {
            const request = new AllstaskRequest(HttpMethod.GET, "/api/v1/batch-jobs/{batch_job_id}");

            request.setPathParam("batch_job_id", batchJobId);
            
                return request.send(
                    this.requestCtx,
                    BatchJobDetailResponseSerializer._fromJsonObject,
                );
            }

        

    /** Retrieve paginated failures for a batch job. */
        public listBatchJobFailures(
            batchJobId: string,
            options?: BatchJobsListBatchJobFailuresOptions,
            ): Promise<BatchJobFailuresResponse> {
            const request = new AllstaskRequest(HttpMethod.GET, "/api/v1/batch-jobs/{batch_job_id}/failures");

            request.setPathParam("batch_job_id", batchJobId);
            request.setQueryParam("chunk_id", options?.chunkId);
            request.setQueryParam("limit", options?.limit);
            request.setQueryParam("offset", options?.offset);
            
                return request.send(
                    this.requestCtx,
                    BatchJobFailuresResponseSerializer._fromJsonObject,
                );
            }

        

    }

